src/paperless/settings.py

```diff
@@ -409,8 +415,38 @@ def _parse_beat_schedule() -> dict:
 AUTHENTICATION_BACKENDS = [
     "guardian.backends.ObjectPermissionBackend",
     "django.contrib.auth.backends.ModelBackend",
+    "allauth.account.auth_backends.AuthenticationBackend",
 ]
 
+SOCIALACCOUNT_PROVIDERS = {
+    # "openid_connect": {
+    #    "APPS": [
+    #        {
+    #            "provider_id": "keycloak",
+    #            "name": "Keycloak",
+    #            "client_id": "paperless",
+    #            "secret": "<CLIENT_SECRET>",
+    #            "settings": {
+    #                "server_url": "https://<KEYCLOAK_SERVER>/realms/<REALM>/.well-known/openid-configuration",
+    #            },
+    #        },
+    #    ],
+    # },
+}
```

Comment:
```suggestion
SOCIALACCOUNT_PROVIDERS = json.loads(os.getenv(
    "PAPERLESS_SOCIALACCOUNT_PROVIDERS",
)) if "PAPERLESS_SOCIALACCOUNT_PROVIDERS" in os.environ else None
```

Presumably should be something like this so users aren't editing settings.py directly, then you can use e.g.

`PAPERLESS_SOCIALACCOUNT_PROVIDERS='{"github": {"APPS": [{"provider_id": "github","name": "Github New","client_id": "xxx","secret": "xxx"}]}}'` in paperless.conf, docker ENV, etc.

Reply:
Sounds good to me. I pushed a change to load the providers from the environment. Do you want a dedicated variable for the provider apps? Or is using the existing `PAPERLESS_APPS` good enough?

Reply:
Cool. I think its fine as-is, tho again I’ll defer to the backend review